src/main/scala/chisel3/stage/phases/Elaborate.scala

```diff
@@ -21,8 +22,20 @@ class Elaborate extends Phase {
   override def invalidates(a: Phase) = false
 
   def transform(annotations: AnnotationSeq): AnnotationSeq = annotations.flatMap {
-    case a: ChiselGeneratorAnnotation => a.elaborate
-    case a                            => Some(a)
+    case a: ChiselGeneratorAnnotation => try {
+      a.elaborate
+    } catch {
+      /* if any throwable comes back and we're in "stack trace trimming" mode, then print an error and trim the stack trace
+       */
+      case a: Throwable =>
+        logger.error("Error during elaboration!")
+        if (!view[ChiselOptions](annotations).printFullStackTrace) {
+          logger.error("Stack trace will be trimmed to user code only. Rerun with --full-stacktrace to see the full stack trace")
+          a.trimStackTraceToUserCode()
+        }
```

Comment:
See comments above, I don't love the `logger.error` here, but this is still sooo much better than the status quo.

Reply:
Will drop the `--full-stacktrace` logger.error in a fixup commit. Were you talking about the `Error during elaboration`, too?

Reply:
I feel less strongly about `Error during elaboration`. I'd probably drop it since an Exception is a pretty good indicator something went wrong, but if you'd rather keep it I think it's fine.

Reply:
Updated in rebase. Let me know if you want the earlier logger.error removed.

Reply:
(GitHub didn't refresh for the above comment.) Changed in a fixup commit on both branches.